connaisseur/validate.py

```diff
@@ -124,14 +128,29 @@ def process_chain_of_trust(host: str, image: Image, req_delegations: list):
     # required delegation JSON's. otherwise take the targets field of the targets JSON, as
     # long as no delegations are defined in the targets JSON. should there be delegations
     # defined in the targets JSON the targets field of the releases JSON will be used.
+    # unfortunately there is a case, where delegations could have been added to a
+    # repository, but no signatures were created using the delegations. in this special
+    # case, the releases JSON doesn't exist yet and the targets JSON must be used instead
     if req_delegations:
+        for target_role in req_delegations:
+            if not trust_data[target_role]:
+                raise NotFoundException(
+                    'no trust data for delegation role "{}" for image "{}"'.format(
+                        target_role, str(image)
+                    ),
+                    {"tuf_role": target_role},
+                )
+
```

Comment:
You can make this a bit more neat as

```python
if not all(trust_data[target_role] for target_role in req_delegations):
    tuf_roles = [target_role in req_delegations if not trust_data[target_role]]
    msg = f"no trust data for delegation roles {tuf_roles} for image {image}"
    raise NotFoundException(msg, {"tuf_roles": tuf_roles})
```

or, weirder,

```python
if tuf_roles := [target_role in req_delegations if not trust_data[target_role]]:
    msg = f"no trust data for delegation roles {tuf_roles} for image {image}"
    raise NotFoundException(msg, {"tuf_roles": tuf_roles})
```

I think nobody would understand the second, and I really dislike inline assignments like that.